test/iscsi_client_lib_test.rb

```diff
@@ -16,54 +14,60 @@
       allow(subject.log).to receive(:error)
     end
 
-    context "when iscsid.socket and iscsid.service are active" do
+    context "for iscsid.socket" do
+      let(:iscsid_socket) { double("iscsid.socket", start: true) }
+
       before do
-        allow(Yast::Service).to receive(:active?).with("iscsid").and_return(true)
         allow(Yast::SystemdSocket).to receive(:find!).with("iscsid").and_return(iscsid_socket)
-        allow(subject).to receive(:socketActive?).with(iscsid_socket).and_return(true)
+        allow(subject).to receive(:socketActive?).with(iscsid_socket).and_return(socket_status)
       end
 
-      it "does nothing" do
-        expect(iscsid_socket).to_not receive(:start)
-        expect(Yast::Service).to_not receive(:Stop).with("iscsid")
-
-        subject.getServiceStatus
-      end
-    end
+      context "when iscsid.socket and iscsid.service are active" do
+        let(:socket_status) { true }
 
-    context "when iscsid.socket is not active" do
-      before do
-        allow(Yast::SystemdSocket).to receive(:find!).with("iscsid").and_return(iscsid_socket)
-        allow(subject).to receive(:socketActive?).with(iscsid_socket).and_return(false)
-      end
+        before do
+          allow(Yast::Service).to receive(:active?).with("iscsid").and_return(true)
+        end
 
-      context "and socket is availbale" do
-        it "start iscsid.socket" do
-          expect(iscsid_socket).to receive(:start)
+        it "does nothing" do
```

Comment:
I would say something like "does not start the socket nor the service". But *does nothing* says nothing to me :)

Reply:
Okay, thank you :) :)

I will to reuse your proposition :stuck_out_tongue_winking_eye: